pulp_smash/tests/rpm/api_v2/test_remove_unit.py

```diff
@@ -20,162 +8,192 @@
+        on_demand_ids = _get_rpm_ids(
+            _get_rpms(self.cfg, self.repos['on demand'])
         )
-        cls.responses['third sync'] = utils.sync_repo(cls.cfg, hrefs[1])
-
-        # Search for units in both repositories again
-        for i, href in enumerate(hrefs):
-            cls.responses['repo {} units, post'.format(i)] = client.post(
-                urljoin(href, 'search/units/'),
-                {'criteria': {}},
-            )
-
-    def test_status_code(self):
-        """Verify the HTTP status code of each server response."""
-        for step, code in (
-                ('first sync', 202),
-                ('second sync', 202),
-                ('third sync', 202),
-                ('first publish', 202),
-                ('second publish', 202),
-                ('repo 0 units, pre', 200),
-                ('repo 1 units, pre', 200),
-                ('repo 0 units, post', 200),
-                ('repo 1 units, post', 200),
-                ('distribute', 201),
-                ('remove unit', 202),
-        ):
-            with self.subTest(step=step):
-                self.assertEqual(self.responses[step].status_code, code)
-
-    def test_units_before_removal(self):
-        """Assert the repositories have the same units before the removal.
-
-        Package category and package group differ, so we count only RPM units.
+        self.assertEqual(root_ids, on_demand_ids)
+
+    def test_03_update_root_repo(self):
+        """Remove a content unit from the root repository and republish it."""
+        unit = random.choice(_get_rpms(self.cfg, self.repos['root']))
+        criteria = {
+            'filters': {'unit': {'name': unit['metadata']['name']}},
+            'type_ids': [unit['unit_type_id']],
+        }
+        api.Client(self.cfg).post(
+            urljoin(self.repos['root']['_href'], 'actions/unassociate/'),
+            {'criteria': criteria}
+        ).json()
+        utils.publish_repo(self.cfg, self.repos['root'])
+
+        # Verify the removed unit cannot be found via a search.
+        units = find_units(self.cfg, self.repos['root'], criteria)
+        self.assertEqual(len(units), 0, units)
+
+    def test_04_sync_immediate_child(self):
+        """Sync the "immediate" repository.
+
+        Verify it has the same contents as the root repository.
         """
-        self.assertEqual(
-            _get_rpm_ids(self.responses['repo 0 units, pre'].json()),
-            _get_rpm_ids(self.responses['repo 1 units, pre'].json()),
+        utils.sync_repo(self.cfg, self.repos['immediate']['_href'])
+        root_ids = _get_rpm_ids(_get_rpms(self.cfg, self.repos['root']))
+        immediate_ids = _get_rpm_ids(
+            _get_rpms(self.cfg, self.repos['immediate'])
         )
+        self.assertEqual(root_ids, immediate_ids)
 
-    def test_units_after_removal(self):
-        """Assert the repositories have the same units after the removal.
+    def test_04_sync_on_demand_child(self):
```

Comment:
The full list of methods is:

```
test_01_create_root_repo
test_02_create_immediate_child
test_02_create_on_demand_child
test_03_update_root_repo
test_04_sync_immediate_child
test_04_sync_on_demand_child
```

The numbering basically indicates that the `02*` and `04*` steps can happen in any order, and even in parallel. If unittest offered a mechanism for running select test methods in parallel, that would be cool...